docs/integration-testing.md

```diff
@@ -0,0 +1,86 @@
+# Integration Testing
+
+Integration testing is the phase in software testing in which individual software modules are combined and tested as a group. 
+Integration testing is conducted to evaluate the compliance of a system or component with specified functional requirements. 
+It occurs after unit testing and before system testing. Integration testing takes as its input modules that have been unit tested, groups them in larger aggregates, applies tests defined in an integration test plan to those aggregates, and delivers as its output the integrated system ready for system testing.
+
+To group these integration test aggregates, we make use of test suites. In software development, a test suite, less commonly known as a validation suite, is a collection of test cases that are intended to be used to test a software program to show that it has some specified set of behaviours. 
+A test suite often contains detailed instructions or goals for each collection of test cases and information on the system configuration to be used during testing. 
+A group of test cases may also contain prerequisite states or steps, and descriptions of the following tests.
+
+## Database Integration Testing (Postgres)
+
+To run our test suite we use the go library [testify](https://pkg.go.dev/github.com/stretchr/testify/suite), specifically the suite capability. 
+Compared to regular testing, this enables us to manage lifecycle of tests run inside a suite through a set of predefined methods (for example, run before or after all or individual tests are executed). 
+As a result, it enables us to manage dependant components of the reconciler, in this case our database.
+
+### How to write a Postgres Integration Test
```

Comment:
For your headlines, you're switching between title case and sentence case. 
[Kyma guidelines](https://kyma-project.io/community/guidelines/content-guidelines/03-formatting/#headings) want sentence case, [SAP guidelines](https://help.sap.com/viewer/DRAFT/e33c591ae4494a659a3f5f983c9d1161/PROD/en-US/1c06558f75ef45c686c83e6e73195142.html) want title case. As this is internal documentation, pick what you like as long as it's consistent.

Reply:
```suggestion
### How to write a Postgres integration test
```